tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -525,35 +545,45 @@ class CommandQueue {
     // Blocking if in passthrough mode. Non-blocking if in async mode
     void run_command(const CommandInterface& command);
 
-    // API for setting the mode of the command queue
-    // The environment env counterpart is `TT_METAL_ASYNC_QUEUES`
+    // API for setting/getting the mode of the command queue
     void set_mode(const CommandQueueMode& mode);
+    CommandQueueMode get_mode() const { return this->mode; }
 
     // Reference to the underlying hardware command queue, non-const because side-effects are allowed
     HWCommandQueue& hw_command_queue();
 
+    // The empty state of the worker queue
+    bool empty() const { return this->worker_queue.empty(); }
+
    private:
     enum class CommandQueueState {
         IDLE = 0,
         RUNNING = 1,
         TERMINATE = 2,
     };
-    CommandQueueMode mode = CommandQueue::get_mode();
-    CommandQueueState worker_state = CommandQueueState::IDLE;
+    constexpr static uint32_t TRACE_QUEUE_CQ_ID = 401;
+    friend class Trace;
+    friend void EnqueueTraceImpl(CommandQueue& cq);
+    friend uint32_t InstantiateTrace(Trace& trace, CommandQueue& cq);
+
+    CommandQueueMode mode;
+    CommandQueueState worker_state;
     std::unique_ptr<std::thread> worker_thread;
-    LockFreeQueue<CommandInterface> worker_queue;
+    WorkerQueue worker_queue;
     uint32_t cq_id;
     Device* device_ptr;
+    Trace* trace_ptr;
```

Comment:
This is temporary for implementing the stub functionality, the actual instantiate should not need the trace ptr assignment into the CQ, right now since we don't copy the traced commands as a buffer into DRAM we do need it for the replay and it's a borrow rather than owned hence a raw pointer.

The proper implementation for InstantiateTrace is filed as an issue #5830